interfaces/builtin/adb_support.go

```diff
@@ -0,0 +1,205 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const adbSupportSummary = `allows access to android devices for fastboot and adb`
```

Comment:
`allows USB access to Android ...`

Reply:
Now I'm wondering if the interface shouldn't simply be called 'adb' and if the adb snap should slot this interface and things that plugs adb actually talk to the adb service.

With the current PR, the idea is that adb is provided as an implicit slot for core and classic and that when a snap 'plugs [ adb-support ]' only then are the udev rules added (are we sure they aren't added twice when two snaps plug the interface since we are using connectedPlugUDev?).

It seems that a better representation is that we consider adb as a service, like modem-manager or ofono, that provides the slot. In this manner, an adb snap makes the 'adb' interface available and only when the adb snap is installed are the udev rules added to the system via permanentSlotUDev. Based on the discussion in the forum for UBPorts, it sounds like this is what they are doing: they provide adb in the snap and use it. We then can have plug rules that allow connecting other snaps to the adb slot (and in the case of UBPorts, it would have a command plugs to its slot).

As a convenience on classic, we could, but aren't obligated to do so, provide the implicit classic interface for those systems that happen to have the dev installed. In this case, we would *not* install the udev rules because it is expected that the deb would've done this already.

Until we agree on the approach for the interface, I'll not perform a complete review of the interface.